### PR TITLE
[CPU] Change kvcache type if user sets `INFERENCE_PRECISION_HINT`

### PR DESCRIPTION
**Pull request by luo-cheng2021**
When user sets `INFERENCE_PRECISION_HINT` change the kvcache type.

Ticket:
[145861](https://jira.devtools.intel.com/browse/CVS-145861)

### CODE REVIEW

File: src/cpp/src/device_config.hpp [hunk 1]
```diff
@@ -29,6 +29,20 @@ class DeviceConfig {
         if (m_device == "CPU") {
             auto inference_precision = core.get_property(device, ov::hint::inference_precision);
             m_kv_cache_type = inference_precision == ov::element::bf16 ? ov::element::bf16 : ov::element::f16;
+            // if user sets precision hint, kv cache type should be changed
+            if (plugin_config.find("INFERENCE_PRECISION_HINT") != plugin_config.end()) {
+                const auto& type_name = plugin_config.at("INFERENCE_PRECISION_HINT").as<std::string>();
+                if (type_name == "f32") {
+                    m_kv_cache_type = ov::element::f32;
```
Comment (Wovchena) [inline src/cpp/src/device_config.hpp:36]: Looks like ov bug. It's always supposed to return f32 for floats as far as I know. There's `set_elemnt_type()` to specify a desired type https://github.com/openvinotoolkit/openvino/blob/9cd8ec5455c618427178a0a427ee9107fa83359f/samples/cpp/classification_sample_async/main.cpp#L96

File: src/cpp/src/device_config.hpp [hunk 2]
```diff
@@ -29,6 +29,20 @@ class DeviceConfig {
         if (m_device == "CPU") {
             auto inference_precision = core.get_property(device, ov::hint::inference_precision);
             m_kv_cache_type = inference_precision == ov::element::bf16 ? ov::element::bf16 : ov::element::f16;
+            // if user sets precision hint, kv cache type should be changed
+            if (plugin_config.find("INFERENCE_PRECISION_HINT") != plugin_config.end()) {
+                const auto& type_name = plugin_config.at("INFERENCE_PRECISION_HINT").as<std::string>();
```
Comment (ilya-lavrenov) [inline src/cpp/src/device_config.hpp:34]: why don't we use properties and use hardcoded strings? Let's use OV entities

File: src/cpp/src/device_config.hpp (hunk 1 above)
Comment (ilya-lavrenov) [reply src/cpp/src/device_config.hpp:36]: for SPR inference precision is bf16

Comment (Wovchena) [reply src/cpp/src/device_config.hpp:36]: My expectation is that inference precision shouldn't influence in/out tensors dtype in the first place.

Comment (ilya-lavrenov) [reply src/cpp/src/device_config.hpp:36]: it's KV cache tensors, which we define by ourselves.

Standard model i/o tensors definitely don't depend on inference precision.

File: src/cpp/src/device_config.hpp (hunk 2 above)
Comment (ilya-lavrenov) [reply src/cpp/src/device_config.hpp:34]: @luo-cheng2021 ping

Comment (luo-cheng2021) [reply src/cpp/src/device_config.hpp]: Sorry for late, updated.

File: src/cpp/src/device_config.hpp [hunk 3]
```diff
@@ -29,6 +29,20 @@ class DeviceConfig {
         if (m_device == "CPU") {
             auto inference_precision = core.get_property(device, ov::hint::inference_precision);
             m_kv_cache_type = inference_precision == ov::element::bf16 ? ov::element::bf16 : ov::element::f16;
+            // if user sets precision hint, kv cache type should be changed
+            if (plugin_config.find(ov::hint::inference_precision.name()) != plugin_config.end()) {
+                const auto& type_name = plugin_config.at(ov::hint::inference_precision.name()).as<std::string>();
```
Comment (ilya-lavrenov) [inline src/cpp/src/device_config.hpp:34]:
```suggestion
                const auto& type_name = plugin_config.at(ov::hint::inference_precision.name()).as<ov::element::Type>();
```
w/o any string conversions

Comment (luo-cheng2021) [reply src/cpp/src/device_config.hpp]: Done.

**Review by ilya-lavrenov — APPROVED** [review]

**Review by Wovchena — APPROVED** [review]

**Pull request merged**